internal/standalone/standalone.go

```diff
@@ -139,29 +140,39 @@ func currentProcPath() (string, error) {
 	return curProcPath, nil
 }
 
-// findPluginRootDir will attempt to find a plugin directory based on the executing process's file-system path.
-func findPluginRootDir(curProcPath string) (string, bool) {
-	cwd, _ := os.Getwd()
-	if filepath.Base(cwd) == "pkg" {
-		cwd = filepath.Join(cwd, "..")
-	}
+// findPluginDir will attempt to find a plugin directory based on the provided plugin ID.
+// This function will traverse the filesystem from the provided directory until it finds a directory that contains
+// a plugin.json file containing an id matching the provided plugin ID.
+func findPluginDir(dir, pluginID string) (string, bool) {
+	var pluginDir string
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
```

Comment:
does this follow symlinks?

Reply:
No it doesn't actually. Should it? 😄

Reply:
I have my plugins folder full of symlinks, not sure if it's relevant 🤷‍♂️

Reply:
This is really only relevant when trying to debug via standalone mode. I believe people are doing that from within the plugin repo and then maybe having a symlinked folder `data/plugins/$pluginFolder` that points to `$clonedPluginRepo/dist/` - which would be fine.

cc @xnyo as I may be missing something 🤔

Reply:
mmh, I tried the PR and started a debug session for a plugin but it's failing:

```
2025/03/07 16:46:56 Error: plugin directory not found for plugin ID: grafana-datasourcehttpbackend-datasource
{"@level":"debug","@message":"Serving plugin","@timestamp":"2025-03-07T16:46:56.564345+01:00","plugins":["diagnostics","resource","data","stream"]}
This binary is a plugin. These are not meant to be executed directly.
Please execute the program that consumes these plugins, which will
load any plugins automatically
```

I can help debugging this if you are not able to reproduce it.

Reply:
ah, when starting a debug session, the working directory is the `pkg` folder. It seems that we just need to restore these lines 

```go

	if filepath.Base(cwd) == "pkg" {
		cwd = filepath.Join(cwd, "..")
	}
```

Reply:
Must be something more to it as those lines are kept (but before this func is called) 🤔 https://github.com/grafana/grafana-plugin-sdk-go/pull/1255/files#diff-40ccbb62d3d29ff8e20335717ce7b56d32c8a5d1bc8defe4fefcca91fee4fb92R85-R88

Will have a look!

Reply:
Wasn't able to reproduce but finally realized the culprit is that the `datasource-http-backend` ID in `plugin.json` and `main.go` don't match 🙈 

- plugin.json: `example-httpbackend-datasource`
- main.go: `grafana-datasourcehttpbackend-datasource`

I will patch that but wondering if that's enough to not update the logic we have here 🤷

Reply:
good catch 👍  tested and it's now working fine for me